packages/neon-core/src/u/misc.ts

```diff
@@ -78,3 +79,44 @@ export function reverseHex(hex: string): string {
   }
   return out;
 }
+
+interface NeonSerializable {
+  size: number;
+  serialize: () => string;
+}
+type Serializables = number | HexString | NeonSerializable[];
+/**
+ * Calculates the byte size of any supported input following NEO's variable int format.
+ */
+export function getSerializedSize(value: Serializables): number {
+  switch (typeof value) {
+    case "number": {
+      if (value < 0xfd) return 1;
+      else if (value <= 0xffff) return 3;
+      else return 5;
+    }
+    case "object": {
+      if (value instanceof HexString) {
+        const size = value.byteLength;
+        return getSerializedSize(size) + size;
+      } else if (Array.isArray(value)) {
+        const arrayLength = value.length;

```

Comment:
At this point, we dont have to enforce the type checks. No need for array length checks or checking if the interface conforms. The code can work fine with empty arrays and we will just throw if the caller passes something weird.
```
        const innerContentsSize = value
          .map((i) => i.size)
          .reduce((a, b) => a + b, 0);
        return getSerializedSize(value.length) + innerContentsSize;
```